src/test/demultiplexer_node_test.cpp

```diff
@@ -0,0 +1,89 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "../dl_node.hpp"
+#include "../logging.hpp"
+#include "../node.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+using testing::ElementsAre;
+
+class DemultiplexerDLNode : public DLNode {
+public:
+    DemultiplexerDLNode(const std::string& nodeName, const std::string& modelName, std::optional<model_version_t> modelVersion, ModelManager& modelManager, std::unordered_map<std::string, std::string> nodeOutputNameAlias, std::optional<uint32_t> demultiplyCount, const NodeSessionMetadata& meta) :
+        DLNode(nodeName, modelName, modelVersion, modelManager, nodeOutputNameAlias, demultiplyCount.value_or(0)) {
+        this->demultiplexCount = demultiplexCount;
+        // createSession to have source session for fetchResults()
+        std::unique_ptr<NodeSession> nodeSession = createNodeSession(meta, 1);
+        auto emplacePair = nodeSessions.emplace(meta.getSessionKey(), std::move(nodeSession));
+        EXPECT_TRUE(emplacePair.second);
+    }
+    void setFetchResult(InferenceEngine::Blob::Ptr& intermediateResultBlob) {
+        this->intermediateResultBlob = intermediateResultBlob;
+    }
+    using Node::fetchResults;
+    Status fetchResults(NodeSession& nodeSession, SessionResults& nodeSessionOutputs) {
+        const auto& sessionMetadata = nodeSession.getNodeSessionMetadata();
+        const auto sessionKey = sessionMetadata.getSessionKey();
+        BlobMap blobs{{std::string("a"), intermediateResultBlob}};
+        std::pair<NodeSessionMetadata, BlobMap> metaBlobsPair{sessionMetadata, std::move(blobs)};
+        nodeSessionOutputs.emplace(sessionKey, std::move(metaBlobsPair));
+        return StatusCode::OK;
+    }
+
+private:
+    InferenceEngine::Blob::Ptr intermediateResultBlob;
+};
+
+TEST(DemultiplexerTest, CheckDemultipliedBlobs) {
```

Comment:
Is it possible to already full end to end test? Load OVMS with configuration file, run pipeline request and check results?

Reply:
No - next task is required.